lib/openhab/dsl.rb

```diff
@@ -604,12 +604,13 @@ def ensure_states
     #     Item1.average_since(12.hours)
     #   end
     #
-    # @see OpenHAB::Core::Items::Persistence
-    #
-    # @param [Object] service service either as a String or a Symbol
+    # @param [Object] service Persistence service either as a String or a Symbol
     # @yield [] Block executed in context of the supplied persistence service
     # @return [Object] The return value from the block.
     #
+    # @see persistence!
+    # @see OpenHAB::Core::Items::Persistence
+    #
     def persistence(service)
```

Comment:
you should re-implement this in terms of `persistence!` now:

```
old = persistence!(service)
yield
ensure
persistence!(old)
```

Reply:
done